### DISCUSSION TITLE
How to intercept the error return by execute and add code to extension

### DISCUSSION THREAD

**Discussion by zhoujia74** [graphql-java/graphql-java#3424] [General]
Hi, I want to standardize our service's graphql error handling by including a standard 'code' entry to extensions in error. It's easier to do for our code raised error with a customized error handler. But for non schema/parsing related error raised by graphql-java library code the customized error handle is not triggered. Is there a way to still intercept such error and add customization to the result?

One example is null for required input (ID!). graphql-java will throw NonNullableValueCoercedAsNullException during parsing validation. The response is like below. I want to add "code": "BAD_INPUT" to extensions.
`{
  "errors": [
    {
      "message": "Variable 'input' has an invalid value: Field 'someField' has coerced Null value for NonNull type 'ID!'",
      "locations": [
        {
          "line": 1,
          "column": 43
        }
      ],
      "extensions": {
        "classification": "ValidationError"
      }
    }
  ]
}`

**Comment by bbakerman** [comment]
`graphql.execution.instrumentation.Instrumentation#instrumentExecutionResult(graphql.ExecutionResult, graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters, graphql.execution.instrumentation.InstrumentationState)` might be a way - you can get the result before it send out and look in it for errors and then adjust them as you see fit

We dont have a callback that allows you to adjust errors as they are created - but this callback would allow it in a whole of result sense
